src/routes/hydra.ts

```diff
@@ -1,162 +1,209 @@
-      logger.info("Initiating ORY Kratos Login flow because neither a ORY Kratos Login Request nor a valid ORY Kratos Session was found.");
-      let return_to_address = encodeURI(currentLocation);
-      res.redirect(`${config.kratos.browser}/self-service/browser/flows/login?prompt=login&return_to=${return_to_address}`);
-      return
-    }
-    if (!challenge) {
-      logger.error("ORY Hydra Login flow could not be completed because no ORY Kratos Login Request and no ORY Hydra Login Challenge were found in the request.");
-      next(new Error("No ORY Kratos Login Request and no ORY Hydra Login Challenge were found in the request."))
-      return
-    } else {
-      // 1. Parse Hydra challenge from query params
-      // The challenge is used to fetch information about the login request from ORY Hydra.
-      // Means we have just been redirected from Hydra, and are on the login page
-      // We must check the hydra session to see if we can skip login
-      // 2. Call Hydra and check the session of this user
-
-      return hydraAdminEndpoint.getLoginRequest(challenge)
-        .then(({ response, body }) => {
-          // If hydra was already able to authenticate the user, skip will be true and we do not need to re-authenticate
-          // the user.
-          if (body.skip) {
-            // You can apply logic here, for example update the number of times the user logged in...
-            // Now it's time to grant the login request. You could also deny the request if something went terribly wrong
-            // (e.g. your arch-enemy logging in...)
-            let acceptLoginRequest = new AcceptLoginRequest()
-
-            acceptLoginRequest.subject = String(body.subject)
-            logger.info("Accepting ORY Hydra Login Request because skip is true: ", acceptLoginRequest)
-            return hydraAdminEndpoint.acceptLoginRequest(challenge, acceptLoginRequest)
-              .then((hydraResponse: any) => {
-              // All we need to do is to confirm that we indeed want to log in the user.
-              
-              // All we need to do now is to redirect the user back to hydra!
-              res.redirect(hydraResponse.redirect_to)
-            });
-          } else if (kratos_session){
-            // Figuring out the user
-            return kratosPublicEndpoint
-              // We need to know who the user is for hydra
-              .whoami(req as { headers: { [name: string]: string } })
-              .then( ({body, response} ) => {
-                // We need to get the email of the user. We don't want to do that via traits as
-                // they are dynamic. They would be part of the PublicAPI. That's not true
-                // for identity.addresses So let's get it via the AdmninAPI
-                const subject = body.identity.id
-                kratosAdminEndpoint
-                  .getIdentity(body.identity.id) 
-                  .then( ({response}) => {
```

Comment:
This does the same thing twice, we already have access to the identity and it's traits so we don't need to fetch it from the Admin API.